src/acrcssc/azext_acrcssc/helper/_taskoperations.py

```diff
@@ -0,0 +1,568 @@
+def _update_task_yaml(acr_task_client, registry, resource_group_name, task, encoded_task):
+    logger.debug("Entering update_task_yaml for task %s", task.name)
+    try:
+        taskUpdateParameters = acr_task_client.models.TaskUpdateParameters(
+            step=acr_task_client.models.EncodedTaskStepUpdateParameters(
+                encoded_task_content=encoded_task))
+
+        acr_task_client.begin_update(resource_group_name,
+                                     registry.name,
+                                     task.name,
+                                     taskUpdateParameters)
+
+        logger.debug(f"Task {task.name} updated successfully")
+    except HttpResponseError as exception:
+        logger.warning(f"Failed to update task {task.name} in registry {registry.name}: {exception}")
+
+
+def _update_task_schedule(acr_task_client, registry, resource_group_name, cron_expression, dryrun):
+    logger.debug(f"Using cron_expression: {cron_expression}")
+    taskUpdateParameters = acr_task_client.models.TaskUpdateParameters(
+        trigger=acr_task_client.models.TriggerUpdateParameters(
+            timer_triggers=[
+                acr_task_client.models.TimerTriggerUpdateParameters(
+                    name='azcli_defined_schedule',
+                    schedule=cron_expression)
+            ]))
+
+    if dryrun:
+        logger.debug("Dry run, skipping the update of the task schedule")
+        return
+    try:
+        acr_task_client.begin_update(resource_group_name,
+                                     registry.name,
+                                     CONTINUOUSPATCH_TASK_SCANREGISTRY_NAME,
+                                     taskUpdateParameters)
+        print("Schedule has been successfully updated.")
+    except HttpResponseError as exception:
+        raise AzCLIError(f"Failed to update the task schedule: {exception}")
+
+
+def _delete_task(cmd, registry, task_name, dryrun):
+    logger.debug("Entering delete_task")
+    resource_group = parse_resource_id(registry.id)[RESOURCE_GROUP]
+
+    try:
+        acr_tasks_client = cf_acr_tasks(cmd.cli_ctx)
+        _delete_task_role_assignment(cmd.cli_ctx, acr_tasks_client, registry, resource_group, task_name, dryrun)
+
+        if dryrun:
+            logger.debug(f"Dry run, skipping deletion of the task: {task_name}")
+            return None
+        logger.debug(f"Deleting task {task_name}")
+        LongRunningOperation(cmd.cli_ctx)(
+            acr_tasks_client.begin_delete(
+                resource_group,
+                registry.name,
+                task_name))
+
+    except Exception as exception:
+        raise AzCLIError(f"Failed to delete task {task_name} from registry {registry.name} : {exception}")
+
+    logger.debug(f"Task {task_name} deleted successfully")
+
+
+def _delete_task_role_assignment(cli_ctx, acrtask_client, registry, resource_group, task_name, dryrun):
+    role_client = cf_authorization(cli_ctx)
+    acrtask_client = cf_acr_tasks(cli_ctx)
```

Comment:
Is this necessary as acrtask_client is already passed as a parameter?

Reply:
no, removed